File: android/src/main/java/top/huic/tencent_im_plugin/message/TextMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.TIMGroupAtInfo;
import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMMessage;
import com.tencent.imsdk.v2.V2TIMTextElem;

import java.util.ArrayList;
import java.util.List;

import top.huic.tencent_im_plugin.message.entity.TextMessageEntity;

/**
 * 文本消息节点
 */
public class TextMessageNode extends AbstractMessageNode<V2TIMTextElem, TextMessageEntity> {
    @Override
    public V2TIMMessage getV2TIMMessage(TextMessageEntity entity) {

        // 有@用户或者@所有人则进入分支
        if ((entity.getAtUserList() != null && entity.getAtUserList().size() >= 1) || (entity.getAtAll() != null && entity.getAtAll())) {
            List<String> atList = new ArrayList<>();
            // @所有人
            if (entity.getAtAll() != null && entity.getAtAll()) {
                atList.add(TIMGroupAtInfo.AT_ALL_TAG);
            }

            // @目标用户
            if (entity.getAtUserList() != null) {
                atList.addAll(entity.getAtUserList());
            }
            return V2TIMManager.getMessageManager().createTextAtMessage(entity.getContent(), atList);
        }
        return V2TIMManager.getMessageManager().createTextMessage(entity.getContent());
    }

    @Override
    public String getNote(V2TIMTextElem elem) {
        return elem.getText();
    }

    @Override
    public TextMessageEntity analysis(V2TIMTextElem elem) {
        return new TextMessageEntity(elem);
    }

    @Override
    public Class<TextMessageEntity> getEntityClass() {
        return TextMessageEntity.class;
    }
}

File: ios/Classes/message/TextMessageNode.swift
import ImSDK

//
//  TextMessageNode.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/3/13.
//  文本消息节点
public class TextMessageNode: AbstractMessageNode {
    override func getV2TIMMessage(params: [String: Any]) -> V2TIMMessage {
        let text: String = getParam(params: params, paramKey: "content")!;
        let atUserList: Any? = getParam(params: params, paramKey: "atUserList");
        let atAll: Any? = getParam(params: params, paramKey: "atAll");

        // 有@用户或者@所有人则进入分支
        if (atUserList != nil && !(atUserList is NSNull)) || (atAll != nil && atAll is NSNull && atAll as! Bool) {
            var atList: [String] = [];

            // 追加@的目标
            if atUserList != nil && !(atUserList is NSNull) {
                for item in atUserList as! [String] {
                    atList.append(item);
                }
            }

            // @所有人
            if atAll != nil && atAll is NSNull && atAll as! Bool {
                atList.append(kImSDK_MesssageAtALL);
            }

            return V2TIMManager.sharedInstance().createText(atMessage: text, atUserList: (atList as AnyObject as! NSArray).mutableCopy() as! NSMutableArray)
        }
        return V2TIMManager.sharedInstance().createTextMessage(text);
    }

    override func getNote(elem: V2TIMElem) -> String {
        (elem as! V2TIMTextElem).text ?? "";
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        TextMessageEntity(elem: elem as! V2TIMTextElem)
    }
}
